apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/extensions/NovaSecurityGroupExtension.java

```diff
@@ -135,8 +136,9 @@ public NovaSecurityGroupExtension(NovaApi api,
       }
 
       Set<String> groupNames = instance.getSecurityGroupNames();
-      Set<? extends SecurityGroupInRegion> rawGroups =
-              sgApi.get().list().filter(nameIn(groupNames)).transform(groupToGroupInRegion(region)).toSet();
+       final FluentIterable<org.jclouds.openstack.nova.v2_0.domain.SecurityGroup> allGroups = sgApi.get().list();
+       Set<? extends SecurityGroupInRegion> rawGroups =
+              allGroups.filter(nameIn(groupNames)).transform(groupToGroupInRegion(allGroups, region)).toSet();
```

Comment:
We do need the variable, as it is used both as the root of the iteration and as a parameter to it (so that all groups have knowledge of all other groups).